magicblock-committor-service/src/intent_executor/mod.rs

```diff
@@ -415,6 +573,193 @@ where
             );
         }
     }
+
+    pub async fn prepare_and_execute_strategy<P: IntentPersister>(
+        &self,
+        transaction_strategy: &mut TransactionStrategy,
+        persister: &Option<P>,
+    ) -> IntentExecutorResult<
+        IntentExecutorResult<Signature, TransactionStrategyExecutionError>,
+        TransactionPreparatorError,
+    > {
+        // Prepare message
+        let prepared_message = self
+            .transaction_preparator
+            .prepare_for_strategy(
+                &self.authority,
+                transaction_strategy,
+                persister,
+            )
+            .await?;
+
+        // Execute strategy
+        let execution_result = self
+            .execute_message_with_retries(
+                prepared_message,
+                &transaction_strategy.optimized_tasks,
+            )
+            .await;
+
+        Ok(execution_result)
+    }
+
+    /// Attempts and retries to execute strategy and parses errors
+    async fn execute_message_with_retries(
+        &self,
+        prepared_message: VersionedMessage,
+        tasks: &[Box<dyn BaseTask>],
+    ) -> IntentExecutorResult<Signature, TransactionStrategyExecutionError>
+    {
+        const RETRY_FOR: Duration = Duration::from_secs(2 * 60);
```

Comment:
I'm not sure retrying for up to 2mins in the middle here is a good idea.
Why not fail properly and add this as a pending task to the db and have it retried again via the main retry task?

Also be aware while you're retrying here and the validator stops (which is likely within 2mins) the pending executions would get lost. Therefore you need to write them to the db in itially anyways (I hope we're doing that).